### PR TITLE
Add a Perl6::SysConfig class

### PR DESCRIPTION
**Pull request by patrickbkr**
This PR in combination with [Raku/nqp#653](https://github.com/Raku/nqp/pull/653) will hopefully fix [rakudo/rakudo#3824](https://github.com/rakudo/rakudo/issues/3824) for good.

I'm still pondering whether `rakudo-build-config` should just be left out of `SysConfig` or not.

I do value feedback on this, as I'm not entirely convinced this is the best way to actually solve #3824.

### CODE REVIEW

File: src/Perl6/Compiler.nqp [hunk 1]
```diff
@@ -9,6 +9,10 @@ class Perl6::Compiler is HLL::Compiler {
     has $!can_language_versions; # List of valid language version
     has $!rakudo-home;
 
+    method config() {
```
Comment (vrurg) [inline src/Perl6/Compiler.nqp:12]: Same note as https://github.com/Raku/nqp/pull/653/files#r464615285

File: src/main.nqp [hunk 2]
```diff
@@ -14,8 +17,8 @@ $comp.parsegrammar(Perl6::Grammar);
 $comp.parseactions(Perl6::Actions);
 $comp.addstage('syntaxcheck', :before<ast>);
 $comp.addstage('optimize', :after<ast>);
-hll-config($comp.config);
-nqp::bindhllsym('Raku', '$COMPILER_CONFIG', $comp.config);
+hll-config(nqp::gethllsym('default', 'SysConfig').rakudo-build-config);
+nqp::bindhllsym('Raku', '$COMPILER_CONFIG', nqp::gethllsym('default', 'SysConfig').rakudo-build-config);
```
Comment (vrurg) [inline src/main.nqp:21]: Similar note as for `config()` method name. Except that here a name like `$SYSTEM_CONFIG` fits even better than `$COMPILER_SYSCONFIG`.

**Comment by vrurg** [comment]
Other than the minor doubts I, both PRs looks promising to me.

File: src/Perl6/Compiler.nqp (hunk 1 above)
Comment (patrickbkr) [reply src/Perl6/Compiler.nqp]: Done.

File: src/main.nqp (hunk 2 above)
Comment (patrickbkr) [reply src/main.nqp]: I removed `$COMPILER_SYSCONFIG` entirely. It was only used in a few places.

**Comment by timo** [comment]
it randomly occurs to me: do we make sure to escape stuff properly when substituting into the .in files? If a path contains a ' in it, we could be in trouble ...

**Comment by patrickbkr** [comment]
@timo: There are a set of macros in `nqp-configure/lib/NQP/Macros.pm` that do quoting for different situations, (Makefile depenency, sh, CMD, C source, maybe more) which should be used at the place of use in the *.in files.

I think there are some chars that can't be escaped in Makefiles (spaces?) for which the only solution is to put the thing into a Makefile variable, but that's not a concern of the templating system.

**Comment by patrickbkr** [comment]
I'm going to merge this and the respective NQP PR now, so we'll have some days of testing left before the next release. If problems show up we'll have some time to react.

**Pull request merged**